evm/deploy-gateway-v5.0.x.js

```diff
@@ -341,10 +366,63 @@ async function upgrade(config, options) {
         if (anwser !== 'y') return;
     }
 
-    const tx = await gateway.upgrade(contractConfig.implementation, implementationCodehash, setupParams, gasOptions);
-    printInfo('Upgrade transaction', tx.hash);
+    // Offline signing
+    let gasLimit, gasPrice;
 
-    await tx.wait(chain.confirmations);
+    try {
+        gasPrice = parseUnits((await provider.getGasPrice()).toString(), 'gwei');
+        const block = await provider.getBlock('latest');
+        gasLimit = block.gasLimit.toNumber() / 1000;
+
+        printInfo('Gas Price:', gasPrice.toString());
+        printInfo('Gas Limit:', gasLimit.toString());
+    } catch (error) {
+        printError('Gas price and limit could not be fetched from provider');
+        printObj(error);
+    }
+
+    let nonce = await getNonceFromProvider(provider, signerAddress);
```

Comment:
Can you add a small script that fetches the current nonce for a given list of addresses and updates it in a nonces.json file? `ledgerSign` can just try reading for the nonce from this file for a signer. we do this in the [old scripts](https://github.com/axelarnetwork/key-management-scripts/blob/main/contracts/offline-sign.js#L842)

Reply:
Would be useful to do this for [gas price](https://github.com/axelarnetwork/key-management-scripts/blob/main/contracts/offline-sign.js#L94) and gasLimit as well.   Could be read from the chain config as well temporarily, but I'd like to keep it separate as a next step since we need to keep a high enough gas price/limit to avoid hitting usual variances in the networks